### PR TITLE
[ENH] Add __main__.py and entrypoint for checking version info.

### PR DESCRIPTION
**Pull request by jwlodek**
## Description
Adding `__main__.py` with simple main entrypoint that prints version, update `pyproject.toml`.

## Motivation and Context

Solves #1667

## How Has This Been Tested?

```
(venv) [jwlodek@vdi-10-65-15-187 bluesky]$ which python
~/Workspace/bluesky/venv/bin/python
(venv) [jwlodek@vdi-10-65-15-187 bluesky]$ python --version
Python 3.11.5
(venv) [jwlodek@vdi-10-65-15-187 bluesky]$ python -m bluesky --version
1.13.0a4.dev31+g6168e9d0.d20240415
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -96,6 +96,7 @@ all = ["bluesky[dev,ipython,zmq,common,tools,streamz,plotting,cmd,olog]"]
 
 [project.scripts]
 bluesky-0MQ-proxy = "bluesky.commandline.zmq_proxy:main"
+bluesky = "bluesky.__main__:main"
```
Comment (coretl) [inline pyproject.toml:99]: I wouldn't add it as a script, just so we can do `python -m bluesky -v`

File: src/bluesky/__main__.py
Comment (coretl) [inline src/bluesky/__main__.py:1]: Please can you add a test like this to test:
https://github.com/DiamondLightSource/python-copier-template-example/blob/main/tests/test_cli.py

**Comment by prjemian** [comment]
`python -m setuptools_scm` gives the version. Why is more code necessary?

**Comment by coretl** [comment]
> `python -m setuptools_scm` gives the version. Why is more code necessary?

To make it extensible for the future? Also because argparse makes it easy this way...

**Comment by prjemian** [comment]
> To make it extensible for the future? Also because argparse makes it easy this way...

Only adds code that duplicates existing functions.  Please do not add this now.  If it is wanted in the future, reconsider at that time.

**Comment by prjemian** [comment]
Ok, I see the reason. But 100% agree with not adding a `bluesky` script just for this convenience.

**Comment by jwlodek** [comment]
It seems that the python 3.11 test failure is unrelated?

**Comment by coretl** [comment]
> It seems that the python 3.11 test failure is unrelated?

yes, #1707

**Review by coretl — APPROVED** [review]

**Pull request merged**